cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/cube/DefaultCube.java

```diff
@@ -67,19 +68,26 @@ public DefaultCube(int[] resolutions, FactTableSupplier factTableSupplier,
 
   @Override
   public void add(CubeFact fact) throws Exception {
-    List<Fact> facts = Lists.newArrayList();
-    for (Aggregation agg : aggregations) {
-      if (agg.accept(fact)) {
-        List<TagValue> tagValues = Lists.newArrayList();
-        for (String tagName : agg.getTagNames()) {
-          tagValues.add(new TagValue(tagName, fact.getTagValues().get(tagName)));
+    add(ImmutableList.of(fact));
+  }
+
+  @Override
+  public void add(Collection<CubeFact> facts) throws Exception {
+    List<Fact> toWrite = Lists.newArrayList();
```

Comment:
yeah - spent time picking the right name. I think the one I used is least confusing though :)